### PR TITLE
Postgres dedupe behaves correctly now.

### PR DESCRIPTION
**Pull request by nyarly**

### CODE REVIEW

**Comment by sous-machine-account** [comment]
LGTM

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: util/restful/router.go [hunk 1]
```diff
@@ -60,8 +61,15 @@ type (
...
-			// need to log this here, or we lose the real stack
-			xlog.LogSink.Warnf("%+#v\n%s", p, string(debug.Stack()))
+			if pe, is := p.(error); is {
+				url := "<unknown>"
+				if xlog.Request != nil {
+					url = xlog.Request.RequestURI
+				}
+				logging.ReportError(xlog.LogSink, errors.Wrapf(pe, "%q\n%s", url, string(debug.Stack())))
+			} else {
+				messages.ReportLogFieldsMessage("Panic while processing request", logging.WarningLevel, xlog.LogSink, p, xlog.Request)
```
Comment (JC1738) [inline util/restful/router.go]: Any size consideration for logging xlog.Request?  Now new code will trace out the truncated message to stderr so singularity will pick it up, so should be fine if it is too large will find out.

Comment (nyarly) [reply util/restful/router.go]: The URL and stacktrace should be within the limit. We log more than that routinuely.

The whole Request is probably pushing it though... Ultimately, I don't think that branch ever gets followed, though.

File: util/sqlgen/fieldset_test.go [hunk 2]
```diff
@@ -27,3 +27,22 @@ func TestRowDedupe(t *testing.T) {
 		t.Errorf("Expected first value to be 1, got %v", vals[1])
 	}
 }
+
+func TestRowDedupe_without_candidates(t *testing.T) {
```
Comment (samsalisbury) [inline util/sqlgen/fieldset_test.go]: A test combining candidate with non-candidate fields would be useful as well, testing additional behaviour.

**Pull request merged**